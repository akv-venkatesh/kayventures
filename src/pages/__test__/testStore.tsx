import { applyMiddleware, createStore } from 'redux';
import rootReducer from '../../reducers';
import createSagaMiddleware from '@redux-saga/core';
import { composeWithDevTools } from 'redux-devtools-extension';

export  const testStore = (initialState:any) => {
    const sagaMiddleware = createSagaMiddleware();
    return createStore(
        rootReducer,
        composeWithDevTools(applyMiddleware(sagaMiddleware))
      );
};

export const findByTestAtrr = (component:any, attr:any) => {
    console.log(component);
    const wrapper = component.getByTestId(attr);
    return wrapper;
};