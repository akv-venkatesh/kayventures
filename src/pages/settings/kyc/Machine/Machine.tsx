import React, { Component } from 'react'
import { Button, Col, Container, Form, Modal, Row } from 'react-bootstrap';
import "./machine.css";

import Vector3 from "../../../../assets/icons/various/Vector3.svg"
import MachineIcon from "../../../../assets/icons/various/MachineIcon.svg"
import BottomIcon from "../../../../assets/icons/various/BottomIcon.svg"
import Vector5 from "../../../../assets/icons/various/Vector5.svg"

import PerfectScrollbar from "react-perfect-scrollbar";
import { RiArrowDropRightLine, RiInformationFill } from 'react-icons/ri';
import "react-perfect-scrollbar/dist/css/styles.css";
import Select from '../../../../component/dropdown_select/slelect';
import { AiFillCaretRight } from 'react-icons/ai';



interface typeState {
    showSummary: boolean,
    showMachine: boolean,
    addMoreEnable: boolean,
    machineType: any,
    brandType: any,
    typeTech: any,
    machineCount: any,
    savedState: any,
    toogleCheck: boolean,
    machineKey: number,
}
interface typeProps {

}

class Machine extends Component<typeProps, typeState> {

    constructor(props: any) {
        super(props);
        this.state = {
            machineKey: 0,
            showSummary: false,
            showMachine: false,
            addMoreEnable: false,
            machineType: [],
            brandType: '',
            typeTech: '',
            machineCount: 0,
            toogleCheck: false,
            savedState: [{
                machineType: null,
                machineBrand: null,
                machineTech: null,
                machineCount: null,
                iotEnable: null,
                savedState: false,
            }],
        }
    }

    handleSummary = () => {
        this.setState({
            showSummary: true,
        });
    }
    hideSummary = () => {
        this.setState({
            showSummary: false
        })
    }
    selectMachine = (event: any) => {
        let value: any = event;
        let machineValue = [...this.state.savedState];
        machineValue[this.state.machineKey] = { ...machineValue[this.state.machineKey], machineType: value.value };
        this.setState({
            savedState: machineValue,
        }, () => {
            console.log(machineValue);
        })
    }
    selectBrand = (event: any) => {
        let value: any = event;
        let barndValue = [...this.state.savedState];
        barndValue[this.state.machineKey] = { ...barndValue[this.state.machineKey], machineBrand: value.value };
        this.setState({
            savedState: barndValue,
        }, () => {
            console.log(barndValue);
        })
    }
    selectTech = (event: any) => {
        let value: any = event;
        let techValue = [...this.state.savedState];
        techValue[this.state.machineKey] = { ...techValue[this.state.machineKey], machineTech: value.value };
        this.setState({
            savedState: techValue,
        }, () => {
            console.log(techValue);
        })
    }
    changeMachineCount = (e: any) => {
        if (e.currentTarget.value || e.currentTarget.value === "") {
            let countValue = [...this.state.savedState];
            countValue[this.state.machineKey] = { ...countValue[this.state.machineKey], machineCount: e.currentTarget.value };
            this.setState({
                savedState: countValue,
            }, () => {
                console.log(countValue);
            })
        }
    }
    handletoggle_checkbox = (e: any) => {
        let checkValue = [...this.state.savedState];
        checkValue[this.state.machineKey] = { ...checkValue[this.state.machineKey], iotEnable: e.currentTarget.checked };
        this.setState({
            savedState: checkValue,
            toogleCheck: true,
        }, () => {
            console.log(checkValue);
        })
    }

    handleSavedMachine = (e: any) => {
        let obj = {
            machineType: null,
            machineBrand: null,
            machineTech: null,
            machineCount: null,
            iotEnable: null,
            savedState: false,
        };
        let arr = this.state.savedState;

        const currentState = arr[this.state.machineKey];
        console.log("currentState=>", currentState);


        let machineKey = this.state.machineKey;
        console.log("MACHINEkEY=>", machineKey);
        if (currentState.machieType !== null) {
            machineKey = machineKey + 1;
            arr.push(obj);
        }
        arr[this.state.machineKey] = { ...arr[this.state.machineKey], savedState: true };
        this.setState({
            addMoreEnable: true,
            savedState: arr,
            machineKey: machineKey,

        }, () => {
            console.log(this.state.savedState);
            console.log(this.state.machineKey);
        })
    }
    handleAddMore = () => {
        this.setState({
            showMachine: true,
        })
    }



    render() {

        const machine = [
            { value: 'Single Needle', label: 'Single Needle' },
            { value: 'Double Needle', label: 'Double Needle' },
            { value: 'Thread Over Look', label: 'Thread Over Look' }
        ]
        const brand = [
            { value: 'Juki', label: 'Juki' },
            { value: 'Pfaff', label: 'Pfaff' },
            { value: 'Brother', label: 'Brother' }
        ]
        const technology = [
            { value: 'Basic', label: 'Basic' },
            { value: 'Process Automated', label: 'Process Automated' },
            { value: 'Computerized', label: 'Computerized' }
        ]

        return (
            <div className="machine main d-flex">
                <div className="leftmenu d-flex flex-column">
                    <h4 className="mb-4">Sewing</h4>
                    <h5 >Machine Type</h5>
                    <div className="menu_dropdown mb-3">
                        <div className="mb-3">
                            <form data-testid="machineType">
                                <label htmlFor="select-machine" hidden>select Machine</label>
                                <Select
                                    name="machinetype"
                                    inputId="select-machine"
                                    options={machine}
                                    width='300px'
                                    position='bottom'
                                    placeholder='Select Machine'
                                    onChange={this.selectMachine}
                                ></Select>
                            </form>
                        </div>
                        <div className="mb-3">
                            <form data-testid="machineBrand" >
                                <label htmlFor="select-brand" hidden>select Brand</label>
                                <Select
                                    name="machinebrand"
                                    inputId="select-brand"
                                    options={brand}
                                    width='300px'
                                    position='bottom'
                                    placeholder='Select Brand'
                                    onChange={this.selectBrand}
                                ></Select>
                            </form>
                        </div>
                        <div className="mb-3">
                            <form data-testid="machineTech" >
                                <label htmlFor="select-tech" hidden>select Technology</label>
                                <Select
                                    name="machinetech"
                                    inputId="select-tech"
                                    options={technology}
                                    width='300px'
                                    position='bottom'
                                    placeholder='Select Technology'
                                    onChange={this.selectTech}
                                ></Select>
                            </form>

                        </div>
                    </div>
                    <h5 className="menu_machine_numbers">Add the No. of machines</h5>
                    <div className="mb-3">
                        <div className="machine_input mb-3">
                            <Form.Control
                                type="text"
                                placeholder="0"
                                onChange={this.changeMachineCount}
                            />
                        </div>
                        <div className="iot_switch d-flex mt-4 mb-4">
                            <p className="mr-5">IOT</p>
                            <Form.Check
                                type="switch"
                                data-testid="custom-element"
                                onChange={this.handletoggle_checkbox}
                            />
                            {this.state.toogleCheck ? <p className="ml-5">Enabled</p> : <p className="ml-5">Disabled</p>}

                        </div>
                        <div className="ms-2 mb-3">
                            <Button
                                className="btn btn-secondary submit"
                                disabled={!this.state.toogleCheck}
                                onClick={this.handleSavedMachine}
                            >
                                Submit
                            </Button>
                        </div>
                        <div className="">
                            <div className="plusIcon d-flex">
                                <button style={{ border: 'none' }} onClick={this.handleAddMore} >
                                    <img src={Vector5} className="image_one" alt="" />
                                </button>
                                <p>  Add More </p>
                            </div>
                        </div>
                    </div>
                </div>
                <div className="rightmenu d-flex flex-column">
                    <div className="facility d-flex">
                        <img src={Vector3} alt="" />
                        <p>Facility 1</p>
                        <p>|</p>
                        <img src={Vector3} alt="" />
                        <p>Facility 2</p>
                        <p>|</p>
                        <img src={Vector3} alt="" />
                        <p>Facility 3</p>
                        <p>|</p>
                    </div>
                    <div className="box py-5 mt-2 position-relative">
                        <div className="scroll pb-3">
                            <PerfectScrollbar>
                                <div className="d-flex flex-wrap pe-4">
                                    <Container>
                                        <Row>
                                            {this.state.savedState.map((machine: any) => {
                                                return machine.machineType !== null && (<Col xs={3} md={12} className="column d-flex" >
                                                    <div className="machine_items">
                                                        <div className="machine_image d-flex">
                                                            <img src={MachineIcon} alt="" />
                                                            <h3>{machine.machineCount}</h3>
                                                        </div>
                                                        <p data-testid="testMachineType">{machine.machineType}</p>
                                                    </div>
                                                    <div className="machine_items_text">
                                                        <p>{machine.machineBrand}</p>
                                                        <p>{machine.machineTech}</p>
                                                    </div>
                                                </Col>)
                                            })
                                            }
                                            {
                                                this.state.addMoreEnable ?
                                                    <div className="plusIcon d-flex">
                                                        <button style={{ border: 'none' }} onClick={this.handleAddMore} >
                                                            <img src={Vector5} className="image_one" alt="" />
                                                        </button>
                                                        <p>  Add More </p>
                                                    </div> : null
                                            }
                                        </Row>
                                    </Container>
                                </div>
                            </PerfectScrollbar>
                            <div className="bottom_text d-flex justify-content-around w-100 position-absolute">
                                {this.state.addMoreEnable ? <div className="d-flex">
                                    <img src={BottomIcon} alt="" />
                                    <h6>Add machinary information to view the summary.</h6>
                                </div> : null}
                                <div className="d-flex">
                                    <button
                                        type="submit"
                                        className="btn btn-next next"
                                        disabled={!this.state.addMoreEnable}
                                    >Next&emsp;<AiFillCaretRight />
                                    </button>
                                </div>
                            </div>
                        </div>
                        <div className="d-flex">
                            <div className="summary">
                                <Button href="#" disabled={!this.state.addMoreEnable} size="sm" onClick={this.handleSummary} data-testid="mySummary">
                                    Summary
                                    <RiArrowDropRightLine />
                                </Button>
                            </div>
                        </div>
                    </div>
                </div>


                <Modal
                    show={this.state.showSummary}
                    size="lg"
                    aria-labelledby="contained-modal-title-vcenter"
                    centered
                    className="machine"
                    backdropClassName="machine"
                    onHide={this.hideSummary}
                    data-testid="someElemInMyModal"
                >
                    <Modal.Header closeButton />
                    <Modal.Body className="">
                        <div className="header_discription d-flex">
                            <RiInformationFill className="info-icon" />
                            <p>
                                Select the filters of Machine, Brand and Technology to view the required summary information.
                            </p>
                        </div>
                        <div className="row">
                            <PerfectScrollbar
                                options={{ suppressScrollY: false, suppressScrollX: true }}
                                onScrollY={(container) =>
                                    console.log(`scrolled to: ${container.scrollTop}.`)
                                }
                            >
                                <div className="my-4 mx-3">
                                    <div className="d-flex">
                                        <div className="model_leftmenu d-flex flex-column">
                                            <div className="mb-3">
                                                <div className="mb-3">
                                                    <Select options={machine} width='300px' position='bottom' placeholder='Select Machine' onChange={() => { }}></Select>
                                                </div>
                                                <div className="mb-3">
                                                    <Select options={brand} width='300px' position='bottom' placeholder='Select Brand' onChange={() => { }}></Select>
                                                </div>
                                                <div className="mb-3">
                                                    <Select options={technology} width='300px' position='bottom' placeholder='Select Technology' onChange={() => { }}></Select>
                                                </div>
                                            </div>
                                        </div>
                                        <div className="model_rightmenu d-flex flex-column">
                                            <div className="scroll pb-3">
                                                <PerfectScrollbar>
                                                    <div className="d-flex flex-wrap pe-4">
                                                        <Container>
                                                            <Row>
                                                                {this.state.savedState.map((machine: any) => {
                                                                    return machine.machineType !== null && (<Col xs={3} md={12} className="column d-flex" >
                                                                        <div className="machine_items">
                                                                            <div className="machine_image d-flex">
                                                                                <img src={MachineIcon} alt="" />
                                                                                <h3>{machine.machineCount}</h3>
                                                                            </div>
                                                                            <p>{machine.machineType}</p>
                                                                        </div>
                                                                        <div className="machine_items_text">
                                                                            <p>{machine.machineBrand}</p>
                                                                            <p>{machine.machineTech}</p>
                                                                        </div>
                                                                    </Col>)
                                                                })
                                                                }
                                                            </Row>
                                                        </Container>
                                                    </div>
                                                </PerfectScrollbar>
                                            </div>
                                        </div>
                                    </div>
                                </div>
                            </PerfectScrollbar>

                        </div>
                    </Modal.Body>
                </Modal>
            </div>
        );
    }
}


export default Machine;

