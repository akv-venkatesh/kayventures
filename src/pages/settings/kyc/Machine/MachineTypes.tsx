import React, { Component } from 'react';
import { Col } from 'react-bootstrap';

import MachineIcon from "../../../../assets/icons/various/MachineIcon.svg"

class MachineTypes extends Component {
    render() {
        return (
            <Col xs={3} md={12} className="column d-flex" >
                {/* <div className="machine_items">
                    <div className="machine_image d-flex">
                        <img src={MachineIcon} alt="" />
                        <h3>{this.props.machineCount}</h3>
                    </div>
                    <p>{this.props.state.machineType}</p>
                </div>
                <div className="machine_items_text">
                    <p>{this.state.brandType}</p>
                    <p>{this.state.typeTech}</p>
                </div> */}
            </Col>
        );
    }
}

export default MachineTypes;