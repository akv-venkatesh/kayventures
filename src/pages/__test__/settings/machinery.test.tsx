
import { render, fireEvent } from '@testing-library/react';

import { MemoryRouter as Router } from 'react-router-dom';
import selectEvent from 'react-select-event';


import Machine from '../../settings/kyc/Machine/Machine';
import { testStore } from '../testStore';
let wrapper: any;

const setup = (initialState = {}) => {
    const store = testStore(initialState);
    const wrapper = render(<Router><Machine /></Router>);
    return wrapper;
}
beforeEach(() => {
    wrapper = setup({});

});


test('test machinery text', () => {
    const machine = wrapper.getByRole('heading', { name: /Add the No. of machines/i });
    expect(machine).toBeInTheDocument;
    const type = wrapper.getByRole('heading', { name: /Machine Type/i });
    expect(type).toBeInTheDocument;

    const machineplace = wrapper.queryAllByPlaceholderText('0');
    expect(machineplace).toBeInTheDocument;
    const role = wrapper.queryAllByText('Add machinary information to view the summary.');
    expect(role).toBeInTheDocument;

    const addText = wrapper.queryAllByText('Add Product Category');
    expect(addText).toBeInTheDocument;

    const initialPageText = wrapper.queryAllByText('Click on the machine type to create the machine inventory');
    expect(initialPageText).toBeInTheDocument;
    expect(wrapper.queryAllByText('IOT')).toBeInTheDocument;

})
//fire events
test('onClick clicked', async () => {
    expect(wrapper.getByRole('button', { name: /Next/i })).toHaveAttribute('disabled');
    expect(wrapper.getByRole('button', { name: /Submit/i })).toHaveAttribute('disabled');
    expect(wrapper.getByTestId('custom-element')).not.toBeChecked();

    //selectbox for machineType
    expect(wrapper.getByTestId('machineType')).toHaveFormValues({ machinetype: '' });
    await selectEvent.select(wrapper.getByLabelText('select Machine'), ['Single Needle']);
    expect(wrapper.getByTestId('machineType')).toHaveFormValues({ machinetype: 'Single Needle' });
    expect(wrapper.getByTestId('selected_element')).toBeInTheDocument;

    //selectbox for machineBrand
    expect(wrapper.getByTestId('machineBrand')).toHaveFormValues({ machinebrand: '' });
    await selectEvent.select(wrapper.getByLabelText('select Brand'), ['Juki']);
    expect(wrapper.getByTestId('machineBrand')).toHaveFormValues({ machinebrand: 'Juki' });

    //selectbox for machineTech
    expect(wrapper.getByTestId('machineTech')).toHaveFormValues({ machinetech: '' });
    await selectEvent.select(wrapper.getByLabelText('select Technology'), ['Basic']);
    expect(wrapper.getByTestId('machineTech')).toHaveFormValues({ machinetech: 'Basic' });


    //check event
    expect(wrapper.queryAllByText('Disabled')).toBeInTheDocument;
    const testcheck0 = wrapper.getByTestId('custom-element');
    fireEvent.click(testcheck0);
    expect(testcheck0).toBeChecked();
    expect(wrapper.queryAllByText('Enabled')).toBeInTheDocument;

    //input box change event
    const input = wrapper.getByPlaceholderText('0');
    fireEvent.change(input, { target: { value: "vinoth" } });
    expect(wrapper.getByDisplayValue(/vinoth/i)).toBeInTheDocument;

    expect(wrapper.getByRole('button', { name: /Submit/i })).not.toHaveAttribute('disabled');

    expect(wrapper.getByRole('button', { name: /summary/i })).toHaveAttribute('disabled');
    expect(wrapper.getByRole('button', { name: /next/i })).toHaveAttribute('disabled');

    const submitButton = wrapper.getByRole('button', { name: /submit/i })
    fireEvent.click(submitButton);
    expect(wrapper.getByRole('button', { name: /summary/i })).not.toHaveAttribute('disabled');
    expect(wrapper.getByRole('button', { name: /next/i })).not.toHaveAttribute('disabled');

    //modelbox
    expect(fireEvent.click(wrapper.getByRole('button', { name: /summary/i }))).toBeInTheDocument;
    expect(wrapper.getByRole('dialog')).toBeInTheDocument;
    expect(wrapper.getByText('Select the filters of Machine, Brand and Technology to view the required summary information.')).toBeInTheDocument;
    //in model box select option
    expect(wrapper.getByTestId('machineTypeFilter')).toHaveFormValues({ machinetypefilter: '' });
    await selectEvent.select(wrapper.getByLabelText('Select Machine'), ['Single Needle']);
    expect(wrapper.getByTestId('machineTypeFilter')).toHaveFormValues({ machinetypefilter: 'Single Needle' });

    expect(wrapper.getByTestId('brandtypeFilter')).toHaveFormValues({ brandtypeFilter: '' });
    await selectEvent.select(wrapper.getByLabelText('Select Brand'), ['Juki']);
    expect(wrapper.getByTestId('brandtypeFilter')).toHaveFormValues({ brandtypeFilter: 'Juki' });

    expect(wrapper.getByTestId('techtypefilter')).toHaveFormValues({ techtypefilter: '' });
    await selectEvent.select(wrapper.getByLabelText('Select Technology'), ['Basic']);
    expect(wrapper.getByTestId('techtypefilter')).toHaveFormValues({ techtypefilter: 'Basic' });

    expect(wrapper.getByTestId('iottypefilter')).toHaveFormValues({ iottypefilter: '' });
    await selectEvent.select(wrapper.getByLabelText('IOT'), ['Enable']);
    expect(wrapper.getByTestId('iottypefilter')).toHaveFormValues({ iottypefilter: 'Enable' });

})


