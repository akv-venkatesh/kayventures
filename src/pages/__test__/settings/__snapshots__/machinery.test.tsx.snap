// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`modal opening 1`] = `
<body>
  <div>
    <div
      class="machine main d-flex"
    >
      <div
        class="leftmenu d-flex flex-column"
      >
        <h5
          class="menu_header"
        >
          Machine Type
        </h5>
        <div
          class="menu_dropdown mb-3"
        >
          <div
            class="mb-3"
          >
            <div
              style="width: 300px;"
            >
              <div
                class=" css-b62m3t-container"
              >
                <span
                  class="css-1f43avz-a11yText-A11yText"
                  id="react-select-8-live-region"
                />
                <span
                  aria-atomic="false"
                  aria-live="polite"
                  aria-relevant="additions text"
                  class="css-1f43avz-a11yText-A11yText"
                />
                <div
                  class=" css-1ks0iez-control"
                >
                  <div
                    class=" css-319lph-ValueContainer"
                  >
                    <div
                      class=" css-14el2xx-placeholder"
                      id="react-select-8-placeholder"
                    >
                      Select Machine
                    </div>
                    <div
                      class=" css-6j8wv5-Input"
                      data-value=""
                    >
                      <input
                        aria-autocomplete="list"
                        aria-controls="react-select-8-listbox"
                        aria-describedby="react-select-8-placeholder"
                        aria-expanded="false"
                        aria-haspopup="true"
                        aria-owns="react-select-8-listbox"
                        autocapitalize="none"
                        autocomplete="off"
                        autocorrect="off"
                        class=""
                        id="needleinput"
                        role="combobox"
                        spellcheck="false"
                        style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
                        tabindex="0"
                        type="text"
                        value=""
                      />
                    </div>
                  </div>
                  <div
                    class=" css-1hb7zxy-IndicatorsContainer"
                  >
                    <span
                      class=" css-43ykx9-indicatorSeparator"
                    />
                    <div
                      aria-hidden="true"
                      class=" css-ithc8m-indicatorContainer"
                    >
                      <svg
                        aria-hidden="true"
                        class="css-tj5bde-Svg"
                        focusable="false"
                        height="20"
                        viewBox="0 0 20 20"
                        width="20"
                      >
                        <path
                          d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                        />
                      </svg>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="mb-3"
          >
            <div
              style="width: 300px;"
            >
              <div
                class=" css-b62m3t-container"
              >
                <span
                  class="css-1f43avz-a11yText-A11yText"
                  id="react-select-9-live-region"
                />
                <span
                  aria-atomic="false"
                  aria-live="polite"
                  aria-relevant="additions text"
                  class="css-1f43avz-a11yText-A11yText"
                />
                <div
                  class=" css-1ks0iez-control"
                >
                  <div
                    class=" css-319lph-ValueContainer"
                  >
                    <div
                      class=" css-14el2xx-placeholder"
                      id="react-select-9-placeholder"
                    >
                      Select Brand
                    </div>
                    <div
                      class=" css-6j8wv5-Input"
                      data-value=""
                    >
                      <input
                        aria-autocomplete="list"
                        aria-controls="react-select-9-listbox"
                        aria-describedby="react-select-9-placeholder"
                        aria-expanded="false"
                        aria-haspopup="true"
                        aria-owns="react-select-9-listbox"
                        autocapitalize="none"
                        autocomplete="off"
                        autocorrect="off"
                        class=""
                        id="react-select-9-input"
                        role="combobox"
                        spellcheck="false"
                        style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
                        tabindex="0"
                        type="text"
                        value=""
                      />
                    </div>
                  </div>
                  <div
                    class=" css-1hb7zxy-IndicatorsContainer"
                  >
                    <span
                      class=" css-43ykx9-indicatorSeparator"
                    />
                    <div
                      aria-hidden="true"
                      class=" css-ithc8m-indicatorContainer"
                    >
                      <svg
                        aria-hidden="true"
                        class="css-tj5bde-Svg"
                        focusable="false"
                        height="20"
                        viewBox="0 0 20 20"
                        width="20"
                      >
                        <path
                          d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                        />
                      </svg>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="mb-3"
          >
            <div
              style="width: 300px;"
            >
              <div
                class=" css-b62m3t-container"
              >
                <span
                  class="css-1f43avz-a11yText-A11yText"
                  id="react-select-10-live-region"
                />
                <span
                  aria-atomic="false"
                  aria-live="polite"
                  aria-relevant="additions text"
                  class="css-1f43avz-a11yText-A11yText"
                />
                <div
                  class=" css-1ks0iez-control"
                >
                  <div
                    class=" css-319lph-ValueContainer"
                  >
                    <div
                      class=" css-14el2xx-placeholder"
                      id="react-select-10-placeholder"
                    >
                      Select Technology
                    </div>
                    <div
                      class=" css-6j8wv5-Input"
                      data-value=""
                    >
                      <input
                        aria-autocomplete="list"
                        aria-controls="react-select-10-listbox"
                        aria-describedby="react-select-10-placeholder"
                        aria-expanded="false"
                        aria-haspopup="true"
                        aria-owns="react-select-10-listbox"
                        autocapitalize="none"
                        autocomplete="off"
                        autocorrect="off"
                        class=""
                        id="react-select-10-input"
                        role="combobox"
                        spellcheck="false"
                        style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
                        tabindex="0"
                        type="text"
                        value=""
                      />
                    </div>
                  </div>
                  <div
                    class=" css-1hb7zxy-IndicatorsContainer"
                  >
                    <span
                      class=" css-43ykx9-indicatorSeparator"
                    />
                    <div
                      aria-hidden="true"
                      class=" css-ithc8m-indicatorContainer"
                    >
                      <svg
                        aria-hidden="true"
                        class="css-tj5bde-Svg"
                        focusable="false"
                        height="20"
                        viewBox="0 0 20 20"
                        width="20"
                      >
                        <path
                          d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                        />
                      </svg>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <h5
          class="menu_machine_numbers"
        >
          Add the No. of machines
        </h5>
        <div
          class="mb-3"
        >
          <div
            class="machine_input ms-2 mb-3"
          >
            <input
              class="form-control"
              placeholder="0"
              type="text"
            />
          </div>
          <div
            class="ms-2 mb-3"
          >
            <button
              class="btn btn-secondary submit btn btn-primary"
              disabled=""
              type="button"
            >
              Submit
            </button>
          </div>
          <div
            class="iot_switch ms-2"
          >
            <p
              class=""
            >
              IOT Enabled
            </p>
            <div
              class="form-switch"
            >
              <input
                class="form-check-input"
                data-testid="custom-element"
                type="checkbox"
              />
            </div>
          </div>
          <div
            class=""
          >
            <button
              class="btn btn-secondary reset_switch  btn btn-primary"
              type="button"
            >
              Reset
            </button>
          </div>
        </div>
      </div>
      <div
        class="rightmenu d-flex flex-column"
      >
        <div
          class="facility d-flex"
        >
          <img
            alt=""
            src="Vector3.svg"
          />
          <p>
            Facility 1
          </p>
          <p>
            |
          </p>
          <img
            alt=""
            src="Vector3.svg"
          />
          <p>
            Facility 2
          </p>
          <p>
            |
          </p>
          <img
            alt=""
            src="Vector3.svg"
          />
          <p>
            Facility 3
          </p>
          <p>
            |
          </p>
        </div>
        <div
          class="box py-5 mt-2"
        >
          <div
            class="scroll pb-3"
          >
            <div
              class="scrollbar-container ps"
            >
              <div
                class="ps__rail-x"
                style="left: 0px; top: 0px;"
              >
                <div
                  class="ps__thumb-x"
                  style="left: 0px; width: 0px;"
                  tabindex="0"
                />
              </div>
              <div
                class="ps__rail-y"
                style="top: 0px; left: 0px;"
              >
                <div
                  class="ps__thumb-y"
                  style="top: 0px; height: 0px;"
                  tabindex="0"
                />
              </div>
            </div>
          </div>
          <div
            class="summary"
          >
            <a
              aria-disabled="true"
              class="btn btn-primary btn-sm disabled"
              data-testid="mySummary"
              role="button"
            >
              Summary
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <g>
                  <path
                    d="M0 0h24v24H0z"
                    fill="none"
                  />
                  <path
                    d="M12.172 12L9.343 9.172l1.414-1.415L15 12l-4.243 4.243-1.414-1.415z"
                  />
                </g>
              </svg>
            </a>
          </div>
        </div>
        <div
          class="bottom_text d-flex justify-content-center"
        >
          <img
            alt=""
            src="BottomIcon.svg"
          />
          <h6>
            Add machinary information to view the summary.
          </h6>
        </div>
      </div>
    </div>
  </div>
</body>
`;
